`make tests` crashes

When running `make tests` I get the following error message:

```
[nodetest            ] ~/CindyJS/CindyJS/ref/js/runtests.js:5
[nodetest            ] global.navigator = {};
[nodetest            ]                  ^
[nodetest            ] 
[nodetest            ] TypeError: Cannot set property navigator of #<Object> which has only a getter
[nodetest            ]     at Object.<anonymous> (~/CindyJS/CindyJS/ref/js/runtests.js:5:18)
[nodetest            ]     at Module._compile (node:internal/modules/cjs/loader:1734:14)
[nodetest            ]     at Object..js (node:internal/modules/cjs/loader:1899:10)
[nodetest            ]     at Module.load (node:internal/modules/cjs/loader:1469:32)
[nodetest            ]     at Function._load (node:internal/modules/cjs/loader:1286:12)
[nodetest            ]     at TracingChannel.traceSync (node:diagnostics_channel:322:14)
[nodetest            ]     at wrapModuleLoad (node:internal/modules/cjs/loader:235:24)
[nodetest            ]     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:151:5)
[nodetest            ]     at node:internal/main/run_main_module:33:47
[nodetest            ] 
[nodetest            ] Node.js v23.11.0
node ref/js/runtests.js exited with code 1
make: *** [tests] Error 1
```

As far as I can tell the problem is that in newer versions of Node.js (apparently since v21: https://github.com/nodejs/node/issues/39540#issuecomment-2614062593) `global.navigator` is a predefined read-only object.

--- a/make/build.js
+++ b/make/build.js
@@ -177,7 +177,11 @@ module.exports = function build(settings, task) {
     //////////////////////////////////////////////////////////////////////
 
     task("nodetest", ["plain"], function () {
-        this.node("ref/js/runtests.js");
+        if (process.version > "v21.2") {
+            this.node("--no-experimental-global-navigator", "ref/js/runtests.js");
+        } else {
+            this.node("ref/js/runtests.js");
+        }
     });
 
     task("tests", ["closure", "nodetest", "unittests", "excomp"]);
@@ -187,7 +191,11 @@ module.exports = function build(settings, task) {
     //////////////////////////////////////////////////////////////////////
 
     task("unittests", ["exposed", "plain"], function () {
-        this.cmdscript("mocha", "tests");
+        if (process.version > "v21.2") {
+            this.cmdscript("mocha", "tests", "--no-experimental-global-navigator");
+        } else {
+            this.cmdscript("mocha", "tests");
+        }
     });
 
     //////////////////////////////////////////////////////////////////////

